### ISSUE TITLE
Add the force rotation, pop out button, and background player buttons back

### ISSUE THREAD

**Issue by Ferrero11** [TeamNewPipe/NewPipe#4414]
### Checklist

- [x] I checked, but didn't find any duplicates of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one feature request. I will open one issue for every feature I want to request.

#### Describe the feature you want
Before update 20.0 there use to be certain buttons in the video player that or now missing. Where the open in youtube button is now, there used to be the background player and pop out button, along with a force video rotation button. The force rotation button was especially useful for watching vertical videos. I know that pop out and background player got moved below the video but it was easier to access these features when watching in full screen mode. Also, the video player doesn't automatically go to full screen in portrait orientation, so if that could be reverted. 

#### Is your feature request related to a problem? Please describe it
These features are not problems, but a matter of convenience. Instead of leaving fullscreen to access them, there were right on the video player.

#### Additional context
Add any other context, like screenshots, about the feature request here

If you add this back I'll buy the devs big Macs.

#### How will you/everyone benefit from this feature?
This is a useful feature for users who watch videos in full screen mode, as most people do. Instead of exiting the video they can make adjustments from there and not ruin the emersion or viewing experience.

**Comment by Stypox** [comment]
About the background and popup buttons, I could agree with you. But why would you need an autorotation button? Now everything is handled automatically: vertical videos are detected as such and their orientation is changes; fullscreen is activated automatically by rotating the device (if autorotation is active); ...

> If you add this back I'll buy the devs big Macs.

;-)

**Comment by Ferrero11** [comment]
I guess if it's all automatic then auto rotate isn't needed but I would still like the background player and pop out buttons back. Also why wouldn't videos automatically go to full screen when played even if the phone isn't horizontal. Who watches video in the default mode where you can see the channel and comments section?

Where do I send the Big Macs?

**Comment by comradekingu** [comment]
@Ferrero11 https://liberapay.com/TeamNewPipe/ :) There is also a Bitcoin address on https://newpipe.schabi.org/donate/

**Comment by Ferrero11** [comment]
When the update comes out I'll buy it as long as it's not to much. How many devs am I buying for?

**Comment by opusforlife2** [comment]
@Ferrero11 I'm here to save the day. 😎 

Regarding pop-up and background buttons:
1. In video details page: they are right there, no issue.
2. Full screen (orientation unlocked): Rotate the phone and use the buttons.
3. Full screen (orientation locked): Swipe up on full screen button and use them.

Regarding force rotation button: Did you read the relevant section in the blog post? If not, make sure you do. The full screen button has been worked on extensively. It can do everything you're asking for.

**Comment by Ferrero11** [comment]
@opusforlife2 I know they are in the video details page but they used to be where the "open in youtube" button is now across from the captions button. This was a useful place for them since they could be accessed in full screen mode without exiting the video. The problem with swiping up from the bottom to access these buttons is that for people who use gestures, swiping up from the bottom activates gestures like google assistant and recent apps sometimes.  Putting them next to the share and mute buttons in the drop down menu would let you keep watching the video while accessing these buttons.

**Comment by opusforlife2** [comment]
Oh no. Another gesture conflict. :/

**Comment by hp8wvvvgnj6asjm7** [comment]
I miss that rotation button too

**Comment by hp8wvvvgnj6asjm7** [comment]
- going from popup to full screen now requires more actions than before

- add option in settings to disable automatic rotation to get that rotation button back

**Comment by hipunk** [comment]
I wish for all those buttons to return: rotation by pressing a button in fullscreen, popout by pressing a button in fullscreen, background player by pressing a button in fullscreen. I used all of those!
With the inline player you can watch vertically, but only in one (the default) aspect ratio.

**Comment by Kusurisan** [comment]
100% agree! i did downgrade again to 1.9 after seeing that they removed the buttons. i always used them & the fact that i now need 2 clicks instead of just 1 to go to fullscreen etc. is just annoying.. before that you could click on the play button on the video page and it did playback the video in fullscreen without having to enable rotation etc etc.. now it's just messed up and annoying.

**Comment by Ferrero11** [comment]
this issue is closed so go to the PR they linked to make comments @hipunk @hp8wvvvgnj6asjm7 @Kusurisan

**Comment by opusforlife2** [comment]
Eh, what? It's open.

**Comment by Ferrero11** [comment]
it says that this issue was closed when the PR was linked

**Comment by opusforlife2** [comment]
> was closed

"will close"

**Comment by blackbox87** [comment]
@Stypox 
> About the background and popup buttons, I could agree with you. But why would you need an autorotation button? Now everything is handled automatically: vertical videos are detected as such and their orientation is changes; fullscreen is activated automatically by rotating the device (if autorotation is active); ...

With the exception that you can no longer watch vertical videos in landscape. And if you're thinking "why would you want to do that?" simply queue up a mix of horizontal and vertical videos, disable auto-rotate and then start watching the videos in landscape. When you get to a vertical video it'll be sideways, so you need to physically rotate your phone to watch that video. You previously didn't need to do this.

**Comment by mrlinnth** [comment]
Can I tagalong this thread and share my 2 cent? I'm hoping for share button on popup player. It took me really long to find the share button at current version.

I also wish that on notification panel the close button is showing at right side like in old version. Currently I have to press the down arrow to find the close button.

**Comment by opusforlife2** [comment]
You can customise the buttons which show for the 5 button row and the 3 button row.

**Comment by mrlinnth** [comment]
Holy cow, didn't know about that feature. That's awesome.

**Comment by opusforlife2** [comment]
Read the blog post. You'll get to know lots of stuff!

**Comment by eladkarako** [comment]
<blockquote><del>These features are not problems, but a matter of convenience.</del> this is seriously impact the usage for impaired or limited motor skills and movement users who relays on this. This is not a beautifying issue. <strong>this is accessibility issue</strong>.</blockquote>

**Comment by avently** [comment]
Guys, I removed all those useless buttons. You can blame me.
You may think they are useful. Yes, maybe. But NOT IN THE PLACE WHERE THEY WERE LOCATED. 

You already have popup and background buttons under the player. Copying them to the player itself is a anti-smart idea in the current UI. Because now the buttons under the player is accessible. The problem is that the buttons is hardly accessible. Yes, this is a place for improvement. And instead of arguing you may give an idea of what way the devs may choose for an easy access to the buttons under the player.

TobiGr suggested to create a center zone in the player which will allow to swipe from bottom to top to show UI under a video such as buttons, comments, and so on.
From my point of view it's the best solution. If you have other ideas share them with us.

**Comment by MD77MD** [comment]
>TobiGr suggested to create a center zone in the player which will allow to swipe from bottom to top to show UI under a video such as buttons, comments, and so on.
From my point of view it's the best solution. If you have other ideas share them with us.

I think this and the new enqueue #4425 sould be rushed ASAP. imo all of these complaints are because people used to use newPipe in certain way and was disrupted by the new update, which is normal... however the problem lies that the new update did not include sufficient alternative tools to balance, perhaps those and other PRs should be rushed in the form of a 0.20.1 update or something like that. either this or we risk a negative response from users and maybe going back to 19.8

**Comment by Ferrero11** [comment]
@avently  If you think about the priority of the buttons, some should be switched. Are you more likely to be watching a video and want to quickly switch to a different player, or to open the video in an external browser? If the devs don't want to add the player buttons back to the drop down menu, then maybe do this to satisfy both parties:

Keep the background player and pop up player buttons where they are in the video description. When you tape the globe to open the video in a browser have it instead open a pop out window like the tempo adjuster. Have three options on this pop out:

1. open in external browser
2. background player
3. pop out player

This will allow the number of buttons to stay the same in both places, and the look will be the same, but the people who wan the buttons back in the drop down will have them there hidden away from the people who don't want them there. My other solution would be adding an option in settings to select which buttons you want in the drop down like you can select with the notification shade player. this would allow each user to choose which buttons you have in that menu. If you dont use certain ones, you wouldn't put them there

**Comment by sanityormadness** [comment]
What I think we have with 0.20.0 is a case where the devs are narrowly focusing on *their* use cases, rather than thinking about accessibility or flexibility. If they didn't like something, they cut it. (See avently's comment in #4152 - https://github.com/TeamNewPipe/NewPipe/issues/4152#issuecomment-686628176 - for an even clearer example than their "useless buttons" comment above).

I don't like a single one of the UI changes in 0.20. Not a single one - I want options to revert *everything*, not just a few things like autoplay. For now, I've gone back to 0.19.8, but that's short term - we all know Google will break NewPipe soon enough, and I'll be forced off 0.19.8 when that happens.

I want to force rotation without turning autorotate on (since I want to read in portrait, not have it shuffle randomly when I'm lying on my side, I keep autorotate off. I don't actually care about the popout button and would really prefer to disable everything related to popouts since I don't use it, but people should have it if they want it.) I don't want videos to play on the details page, or keep playing after I leave them unless I fiddle about for an X button. I want to be able to open in browser from the details page (without autoplay), If I'm in a restricted bandwidth setting (or an unusually UNrestricted bandwidth setting, for that matter), I don't want to have to start playing the video or go all the way to the settings page to be able to change quality.

**Comment by avently** [comment]
>0.20.0 is a case where the devs are narrowly focusing on their use cases

It's not true, you can read my initial PR and related discussion about unified player


>I don't want videos to play on the details page, or keep playing after I leave them unless I fiddle about for an X button.

Disable autoplay and disable background playback (set it to None)

>clearer example

It's clear example of how I feel like when two buttons sets have the same action (inside player and outside player). Popup and background buttons under the player are useful.

> I don't like a single one of the UI changes in 0.20.

Looks like you mad because a couple of features work in different way. That's not a big problem, all of the mentioned issues can be fixed almost easily.

>I want to force rotation without turning autorotate on

And whats the problem with current release? Or you have a tablet layout too? If so, I think someone will make an option to instruct NewPipe to act like a phone instead of tablet in case of rotation mechanism.

> I want to be able to open in browser/"share"/etc from the details page (without autoplay),

I can only agree with this. This is downside of my changes and we search an alternative. There is an issue already which you can use for sending ideas.

> I don't want to have to start playing the video or go all the way to the settings page to be able to change quality.

This will not be changed, it's a new workflow

>
All the GUI changes in 0.20 seem to be about forcing us into a smaller box so that we use the app exactly the same way the devs who made these changes do. Regardless of whether we have good reason to prefer the old version or not.

Nope, it's about easy to use UI and a new workflow

**Comment by wb9688** [comment]
>This will not be changed, it's a new workflow

I disagree. I might have 1080p as default quality as I normally want that at home, but sometimes I might want to watch a video elsewhere and then I don't want to waste my 4G data on 1080p while a lower resolution would be OK.

**Comment by opusforlife2** [comment]
Don't we have "Limit resolution when using mobile data" for that particular use case?

**Comment by wb9688** [comment]
@opusforlife2: Hmm… I didn't know that, does that actually limit the resolution (i.e. I can't switch to a higher one if I want)?

**Comment by opusforlife2** [comment]
Never tried it myself. ¯\\\_(ツ)\_/¯

**Comment by sanityormadness** [comment]
> > clearer example
> 
> It's clear example of how I feel like when two buttons sets have the same action (inside player and outside player). Popup and background buttons under the player are useful.

And see, I *WANT* the options "inside player and outside player". When I enter a video, I don't want to have to fiddle about with things after it's already playing/

> > I don't like a single one of the UI changes in 0.20.
> 
> Looks like you mad because a couple of features work in different way. That's not a big problem, all of the mentioned issues can be fixed almost easily.

They could if you gave the options to revert all the changes, not just some.

> > I want to force rotation without turning autorotate on
> 
> And whats the problem with current release? Or you have a tablet layout too? If so, I think someone will make an option to instruct NewPipe to act like a phone instead of tablet in case of rotation mechanism.

#4152

Where you argue that "making the video automatically rotated to landscape from portrait on play looks ugly" and admit you sound "unpleasant" over it and call it an "anti-feature". It's a *core feature* for me. I made a post directly about it on that issue thread: https://github.com/TeamNewPipe/NewPipe/issues/4152#issuecomment-703945152

> I can only agree with this. This is downside of my changes and we search an alternative. There is an issue already which you can use for sending ideas.

But it wouldn't even be an issue if you hadn't removed the bar where it was.

> > I don't want to have to start playing the video or go all the way to the settings page to be able to change quality.
> 
> This will not be changed, it's a new workflow

And again, based on your personal use cases. You don't use it, don't like it, so it goes - same as your "unpleasant" post in #4152

> All the GUI changes in 0.20 seem to be about forcing us into a smaller box so that we use the app exactly the same way the devs who made these changes do. Regardless of whether we have good reason to prefer the old version or not.
> 
> Nope, it's about easy to use UI and a new workflow

So, the GNOME Shell attitude of cutting usability in the name of "ease of use"?

**Comment by sanityormadness** [comment]
Addendum:

> > I don't want videos to play on the details page, or keep playing after I leave them unless I fiddle about for an X button.
> 
> Disable autoplay and disable background playback (set it to None)

I had to reinstall 0.20.0 to check this before I responded to it. That option does not exist. The closest is "minimise on app switch", which defaults to background player. Now, that's certainly a setting that I want off (rather, "None") since if I'm watching video I want to see it - if I want background audio, I will click for that (which is one of the reasons I hate autoplay-video-by-default).

When I say "leave them", I don't just mean "leave the app", I mean "leave the video." I want the video to *stop completely* when I back out of fullscreen, let alone the details page, not just out of the app.

Having to fiddle about for a small X button is not "easier to use" than tapping "back".

**Comment by avently** [comment]
>I don't want to waste my 4G data on 1080p while a lower resolution would be OK.

Who stops you from changing quality once per the whole session? Is it hard? Is it loads a couple of terabytes or what's the real problem? Tons of video apps doesn't provide this feature and no one cares, everything is fine.

If anyone says that is because of bandwidth I will never believe in this because preloading uses as small data as possible. You're watching a video and trying to save a couple of megabytes once per session? Oh, man

**Comment by avently** [comment]
>The closest is "minimise on app switch"

Yes, this is the option I talked about. 

> They could if you gave the options to revert all the changes, not just some.

>  I want the video to stop completely when I back out of fullscreen, let alone the details page, not just out of the app.

For that you can use NewPipe legacy. Thank you for the conversation but your arguments is not something I think useful.

>So, the GNOME Shell attitude of cutting usability in the name of "ease of use"?

Use i3-wm if you think GNOME is unusable.

**Comment by Ferrero11** [comment]
I still think adding an option in settings to select which buttons are in the drop down menu per users preference would be the best option here.

**Comment by avently** [comment]
@Ferrero11 see, your solution is good but for near future. I think the gesture to swipe to description should be added. After that share button and open in browser should be moved under the video like in a scrollable container. We already discussed it and seems like that idea is great.

**Comment by Ferrero11** [comment]
I agree that the share button and open in browser should be moved to the video description if that's what you mean by "scrollable container", and that swiping to the button from the middle of player will work, but I still think the drop down is more streamlined since you can watch the video while using those buttons.

**Comment by lucanomax** [comment]
For the new UI, specifically, I don't like to play the video while on the same screen with comments. I keep comments disabled, so all i get, when watching videos in vertical mode, is an ugly UI.
Before, the player took all the screen, even in vertical mode, so i don't have unnecessary things in my screen (just black up and down, with my video at center).
Sometimes I prefer to keep phone vertically, just that.

Also the notification when playing a video is kinda useless, if i'm watching it and can pause/fast forward directly from the app.

I reverted to previous release, I hope you take this feedback in mind.

**Comment by avently** [comment]
>I keep comments disabled

Why do you do that? I don't think it will save traffic or improve loading time. 
Do you have similar videos disabled too?

@Stypox @TobiGr maybe in case of disabled comments and similar videos would be better to not display empty view (smile and text) but have opened at launch description of the video like happens after a click on video title. This way more screen space will be useful.

>Also the notification when playing a video is kinda useless

Nope, notification is needed, Android requires it when the player runs in a service. Without it Android will kill the app soon.

**Comment by lucanomax** [comment]
>Why do you do that?

I like to have a more minimal UI, also the less i read about idiots in comments the best is for my mental health.

>similar videos disabled

Yes. I watch only videos I search/from my feed and (again) minimal UI. I would love a "_minimal mode_" in the settings, that cuts everything except videos and basic functions.

>notification is needed, Android requires it when the player runs in a service.

Why in 0.19 there isn't a notification when I play a video? (_I'm talking when watching a video in foreground, not background_)

**Comment by avently** [comment]
>Why in 0.19 there isn't a notification when I play a video?

Because the player worked in a different way (not in a service) which makes possible to not show a notification. But the new player has more possibilities like seamless switching between different player types, background playback when you leave the app and much more. Unfortunately it requires to show the notification

**Comment by lucanomax** [comment]
I guess I need to get used to that. At least you can hide notification in android. I hope isn't too much heavy on battery using this player.

**Comment by nbmrjuhneibkr** [comment]
Manual video rotation control definitely needs to be re-implemented. There was absolutely no reason to remove it.

@sanityormadness 
>All the GUI changes in 0.20 seem to be about forcing us into a smaller box so that we use the app exactly the same way the devs who made these changes do. Regardless of whether we have good reason to prefer the old version or not.

Agree 100%. With v0.20 it's starting to look like the devs are taking decisions based only on their personal limited selection of usage scenarios.

**Comment by hipunk** [comment]
> > I keep comments disabled
> 
> Why do you do that? I don't think it will save traffic or improve loading time.
> Do you have similar videos disabled too?
> 
> @Stypox @TobiGr maybe in case of disabled comments and similar videos would be better to not display empty view (smile and text) but have opened at launch description of the video like happens after a click on video title. This way more screen space will be useful.

Oh yes, in many parts of the world you have contracts with a mobile traffic quota. You can only download a certain amount of gigabytes. Example: Contract allows for 1GB of traffic a month. You downloaded 1GB of data in total on 15th. You will be limited to 64KBit/s until next month 1st. Many such contracts exist.
So then, with 64KBit/s, if you have comments and related enabled: You will get a timeout. Error. Especially on EDGE. Without you don't. And you can click download to download lowest quality audio opus. It will then take about 1.4x of the audio files length in time to download. So a 3 minute video will take roughly 5 minutes to download. Very useful. Needed feature. For poor people. For people with 64KBit/s.


> Don't we have "Limit resolution when using mobile data" for that particular use case?

Yes, but when I'm on my balcony I have less throughput in the Wi-Fi. Quality needs to be adjusted on the fly.  However I'm not on mobile. Quality must be adjusted manually.

> > I don't want to waste my 4G data on 1080p while a lower resolution would be OK.
> 
> Who stops you from changing quality once per the whole session? Is it hard? Is it loads a couple of terabytes or what's the real problem? Tons of video apps doesn't provide this feature and no one cares, everything is fine.
> 
> If anyone says that is because of bandwidth I will never believe in this because preloading uses as small data as possible. You're watching a video and trying to save a couple of megabytes once per session? Oh, man

No. Poor people have less bandwidth. Less expensive cool router. Must look worse, there's no option of having 1080p all the time. But sometimes you still might need it.

**Comment by nbmrjuhneibkr** [comment]
@avently 
>you can use NewPipe legacy. Thank you for the conversation but your arguments is not something I think useful

Please stop. An attitude like this will ruin this app. Listen to the users, not just to yourself. NewPipe exists to give users options, options that are not available in the official client. Restricting user choice and forcing everyone to use the app in one specific way goes against this core principle. This is a community-driven project, not a corporate product like the official YouTube app.

"Just go away and use a different app" is the worst response the dev could give to a user who is facing a problem. By the way, NewPipe Legacy hasn't been updated since July, and is incompatible with the latest Android release.

**Comment by avently** [comment]
@hipunk
Thank you for your comment! Interesting information and the use-case you have.

**Comment by opusforlife2** [comment]
@Stypox You've linked your PR but it addresses only 2 concerns out of 3. Could you include "long press on fullscreen button to lock orientation" so that we can test it? We only need to lock when the fullscreen button is visible. When it isn't, the app behaves as expected.

**Comment by TontyTon** [comment]
I want to give more reasonable reason to add full screen/rotation button. I face this everyday and its very irritating.

I use a low end device whose sensors are not that good and fast. So I have to wait some time for the device orientation to change, and using the manual button would be faster. (I want to keep auto rotate on, as it is convinient for many reasons.)

I think, if possible, you can do like youtube Official app. In the app, if I use manual button to switch to full screen then Auto rotate is less sensitive.

---
Also, It would be great if you could add a toggle in settings to set to automatically start the video in fullscreen (like in older versions). So, after pressing the play button, video automatically starts in full screen.

**Comment by opusforlife2** [comment]
@TontyTon Follow #4152 for the second part.

**Comment by streaps** [comment]
> fullscreen is activated automatically by rotating the device (if autorotation is active);
it's not working on mz Android 5.0 tablet, quite the opposite: if I accidentally rotate the tablet it's not fullscreen anymore. rotating it back in landscape mode doesn't change anything.

**Comment by Flashwalker** [comment]
I wote for it

**Comment by Flashwalker** [comment]
Why devs always destroing our happy UX...

**Comment by Flashwalker** [comment]
> fullscreen is activated automatically by rotating the device (if autorotation is active);

Sometimes autorotation fails It is not very reliable to rely only on this sensor or someone else's code.

**Comment by opusforlife2** [comment]
> Why devs always destroing our happy UX...

@Flashwalker Yeah. Silly devs! I wonder who made the happy UX in the first place...

Oh. It was the same devs. xD

**Comment by litetex** [comment]
FYI: 
The background player and popup buttons are still accessible from the fullscreen/landscape player:

Long click the "more options" button
https://user-images.githubusercontent.com/40789489/140622614-0e775dec-6be3-45e4-a638-4baf39218246.mp4

Note: I found this by accident, maybe the UI should me more intuitive.

<sub>Can I get a "Big Mac" now?</sub>

**Comment by avently** [comment]
@litetex 
> Note: I found this by accident, maybe the UI should me more intuitive.

Feel free to send a PR

**Comment by opusforlife2** [comment]
@litetex @avently Is it possible to show tooltips for in-player buttons? We show a tooltip for the background and popup buttons explaining there is a long-press enqueue action available. Maybe something similar could be done for the 'more options' arrow. And maybe the full screen button also, explaining that it can be used to scroll down?

**Comment by avently** [comment]
The solution with a long click was made just to show the possibility to view everything under a video and it was a temporary solution. 

I think more changes should be made in terms of player UI. I tried to add swipe from bottom to top in fullscreen that should trigger scrolling to comments but it looks bad with duration slider at the bottom. 

So I would prefer if the duration slider goes up a little bit and top part of under video view goes up too. This will give the ability to tap and swipe to top. Similar to what official youtube has for mobile and for web in fullscreen.

This will fix the problem with touchable navigation area for Android with gestures and will fix the mentioned problem with swiping to the bottom.

**Comment by opusforlife2** [comment]
What about a drag handle? We could put a small circle at the edge of the screen (on the left or right) which can be grabbed and used to scroll down. I think that will be a simpler solution, requiring much less effort, but being more obvious to the user as well.

We could even add an option in settings on where to place the drag handle, such as top-left, bottom-right, centre-left, etc.

**Comment by avently** [comment]
I don't think that drag handle is more obvious than drag handle + visible part of under player view. And my suggestion is also fixes the problem with a gesture navigation

**Comment by opusforlife2** [comment]
> visible part of under player view

But doesn't this interfere with full screen?

**Comment by avently** [comment]
Nope. Would be challenge to not make an ugly look of bottom player area since I can't imaging a good UI with such elements logic

**Comment by Flashwalker** [comment]
> Nope. Would be challenge to not make an ugly look of bottom player area since I can't imaging a good UI with such elements logic

i can draw non-ugly look for you, good ui is not impossible with my ultra ux/ui design skill

**Comment by comradekingu** [comment]
@Flashwalker It would at least help the people playing at home to see what should be included.
Changing bad design to good design is a lot easier than extracting good design from this thread.
